source/common/common/utility.cc

```diff
@@ -102,6 +102,30 @@ std::string StringUtil::subspan(const std::string& source, size_t start, size_t
   return source.substr(start, end - start);
 }
 
+std::vector<std::string> StringUtil::tokenize(const std::string& source,
```

Comment:
Close, but not quite

split("hello-world", '-') => {"hello", "world"}
tokenize("hello-world", "-") => {"hello", "-", "world"}

If you're cool with an additional argument indicating whether the splitters should be retained or discarded, I can merge them. I'd prefer to do it in a follow-on change to keep the new feature distinct from cleanup.